demos/tutorial/indices/find2_index_approx.cpp

```diff
@@ -0,0 +1,57 @@
+//![Complete]
+#include <set>
+#include <mutex>
+
+#include <seqan/index.h>
+
+using namespace seqan;
+
+int main()
+{
+
+    DnaString genome(
+        "GAGAGGCCACTCGCAGGATTAAGTCAATAAGTTAATGGCGTGGGGTTATGGTATGGGGGTTCTCGCCCACAGTGACCTCATCGGT"
+        "GCATTTCCTCATCGTAGGCGGAACGGTAGACACAAGGCATGATGTCAAATCGCGACTCCAATCCCAAGGTCGCAAGCCTATATAG"
+        "GAACCCGCTTATGCCCTCTAATCCCGGACAGACCCCAAATATGGCATAGCTGGTTGGGGGTACCTACTAGGCACAGCCGGAAGCA");
+    Index<DnaString, BidirectionalIndex<FMIndex<> > > index(genome);
+
+    //![Delegate]
+    auto delegate = [](auto & iter, DnaString const & needle, uint8_t errors)
+    {
+        for (auto occ : getOccurrences(iter))
+            std::cout << occ << std::endl;
+    };
+    //![Delegate]
+
+    DnaString pattern("GGGGTTAT");
+    std::cout << "Hits with up to 2 errors (HammingDistance):" << std::endl;
+    //![SinglePattern]
+    find<0, 2>(delegate, index, pattern, HammingDistance());
+    //![SinglePattern]
+
+    StringSet<DnaString> patterns;
+    appendValue(patterns, "GGGGTTAT");
+    appendValue(patterns, "CTAGCTAA");
+    std::cout << "Hits with 1-2 errors (HammingDistance):" << std::endl;
+    //![MultiplePatterns]
+    find<1, 2>(delegate, index, patterns, HammingDistance(), Serial());
+    //![MultiplePatterns]
+
+    //![ParallelMode]
+    std::mutex mtx;
+    std::set<Pair<DnaString, unsigned> > hits;
+    auto delegateParallel = [&hits, &mtx](auto & iter, DnaString const & needle, uint8_t errors)
```

Comment:
Not important for the PR, but FYI: if you want to have variables specific to the scope of the lambda you can instead:
```cpp
auto delegateParallel = [&hits, mtx = std::mutex{}] (auto & iter, DnaString const & needle, uint8_t errors) mutable
{ ... }
```
This makes it clear that the mutex belongs to the lambda and is not used outside. However specifying the type through assignment and the requirement to make the lambda mutable makes this solution a little less straight-forward.